### PR TITLE
MileStone 8: Integration specs

### PR DESCRIPTION
**Pull request by Efiamotu-1**
> ### ACTIVITIES
> * Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe
> * Integration specs
>   
>   * Use Capybara to write integration tests for each view in your project.
>   * For user index page
>   * For user show page
>   * For user post index page
>   * For user post show page

**Before n + 1**
before-C

before-U

**After n + 1**
after-C
after-U

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Efyamotu, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly ensure that your project adhere(s) to the following standard(s):
- [x] Do not use the bullet gem
- [x] Add screenshots of the logs before and after the fix of the _(N + 1)_ problem.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/integration/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      click_link(@post.title)
+      expect(current_path).to match user_post_path(@user, @post)
+    end
+
+    scenario 'user is redirected to see all posts made by a single user' do
+      click_link('See all posts')
+      sleep(4)
+      expect(current_path).to match user_posts_path(@user)
+    end
+  end
```
Comment (Lucas-Masaba) [inline spec/integration/user_show_spec.rb:63]: - [x] As mentioned in the requirements, please add test(s) that asserts you can see the user's profile picture

File: spec/integration/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    end
+
+    scenario 'display the how many comments' do
+      expect(page).to have_content('Comments: 5')
+    end
+
+    scenario 'display the how many Likes' do
+      expect(page).to have_content('Likes: 0')
+    end
+  end
```
Comment (Lucas-Masaba) [inline spec/integration/post_index_spec.rb:50]: - [x] You are missing the following tests:
- Test(s) that asserts you can see the user's profile picture.
- Test(s) that redirects to the post's show page when you click on a post.

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @Efiamotu-1 and @jdbs9514  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Solved n+1 problems ⚡ 
- Tests are passing 🏄 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cwaku — APPROVED** [review]
Hi @Efiamotu-1 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] All linters passed 🥇 
 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. The previous reviewer simply missed some of them, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**